### DISCUSSION TITLE
Exporting the model to LP or MPS format in Mathopt

### DISCUSSION THREAD

**Discussion by tsjaikishan** [google/or-tools#4327] [Linear Solver questions]
Hello experts,
I am currently building models using the new MathOpt feature in Python. I would like to know how to export a given model to an LP file or MPS file. Previously, with pywraplp, we had the option of using 'solver.ExportModelAsMpsFormat'. Is there a similar feature available in MathOpt?

**Comment by sschnug** [comment]
(Did not touch MathOpt yet)

Following the code, it seem you should be able to that using a 2-step process:

[or-tools/ortools/math_opt/cpp/model.h](https://github.com/google/or-tools/blob/2c333f58a37d7c75d29a58fd772c9b3f94e2ca1c/ortools/math_opt/cpp/model.h#L853)

```cpp
  // Returns a proto representation of the optimization model.
  //
  // See FromModelProto() to build a Model from a proto.
  ModelProto ExportModel(bool remove_names = false) const;
```

[or-tools/ortools/math_opt/io/mps_converter.h](https://github.com/google/or-tools/blob/2c333f58a37d7c75d29a58fd772c9b3f94e2ca1c/ortools/math_opt/io/mps_converter.h#L29)

```cpp
// Returns the model in MPS format.
//
// The RemoveNames() function can be used on the model to remove names if they
// should not be exported.
absl::StatusOr<std::string> ModelProtoToMps(const ModelProto& model);
```

Edit:

Ah... Python...     It seems the exporter is there ([Sources](https://github.com/google/or-tools/blob/2c333f58a37d7c75d29a58fd772c9b3f94e2ca1c/ortools/math_opt/python/model.py#L1812)) but **imho** `ModelProtoToMps` might not be wrapped to python yet (???).

**Comment by rma350** [comment]
Python bindings for that function (MPS reading) were added a few weeks ago, they have not been exported to the open source project yet. The python bindings for reading LP format are still missing.

**Reply by tsjaikishan** [reply]
Could you please let us know approximately how long it will take to see the python bindings for MPS writing/reading in the open-source project?

**Reply by rma350** [reply]
I guess it depends if you are building from head or using pip? If you are using pip, then you need to wait for at least one release. Further, the code isn't exported to github yet. I would guess 1-3 months, but I don't cut new releases.

**Comment by tsjaikishan** [comment]
Thank you @sschnug and @rma350 for your replies. I will wait for the next release 🙂
